tasks/verify/defender.ts

```diff
@@ -0,0 +1,98 @@
+import { execSync } from 'child_process'
+import { task } from 'hardhat/config'
+import { HardhatRuntimeEnvironment as HRE } from 'hardhat/types'
+import { constants } from 'ethers'
+import { appendFileSync } from 'fs'
+import type { VerificationResponse } from '@openzeppelin/hardhat-defender/dist/verify-deployment'
+
+async function main(args: { referenceUrl?: string; contracts: string[] }, hre: HRE) {
+  const { referenceUrl, contracts } = args
+  const { defender, network, graph } = hre
+  const summaryPath = process.env.GITHUB_STEP_SUMMARY
+  if (summaryPath) appendFileSync(summaryPath, `# Contracts deployment verification\n\n`)
+
+  const workflowUrl =
+    referenceUrl ||
+    process.env.WORKFLOW_URL ||
+    execSync(`git config --get remote.origin.url`).toString().trim()
+  const addressBook = graph().addressBook
+  const errs = []
+
+  for (const contractName of contracts) {
+    const entry = addressBook.getEntry(contractName)
+    if (!entry || entry.address === constants.AddressZero) {
+      errs.push([contractName, { message: `Entry not found on address book.` }])
+      continue
+    }
+
+    const addressToVerify = entry.implementation?.address ?? entry.address
+    console.error(`Verifying artifact for ${contractName} at ${addressToVerify}`)
+
+    try {
+      const response = await defender.verifyDeployment(addressToVerify, contractName, workflowUrl)
+      console.error(`Bytecode match for ${contractName} is ${response.matchType}`)
+      if (summaryPath) {
+        appendFileSync(
+          summaryPath,
+          `- ${contractName} at ${etherscanLink(network.name, addressToVerify)} is ${
+            response.matchType
+          } ${emojiForMatch(response.matchType)}\n`,
+        )
+      }
+      if (response.matchType === 'NO_MATCH') {
+        errs.push([contractName, { message: `No bytecode match.` }])
+      }
+    } catch (err: any) {
+      if (summaryPath) {
+        appendFileSync(
+          summaryPath,
+          `- ${contractName} at ${etherscanLink(
+            network.name,
+            addressToVerify,
+          )} failed to verify :x:\n`,
+        )
+      }
+      console.error(`Error verifying artifact: ${err.message}`)
+      errs.push([contractName, err])
+    }
+  }
+
+  if (errs.length > 0) {
+    throw new Error(
+      `Some verifications failed:\n${errs
+        .map(([name, err]) => ` ${name}: ${err.message}`)
+        .join('\n')}`,
+    )
+  }
+}
+
+function etherscanLink(network: string, address: string): string {
+  switch (network) {
```

Comment:
nit: it would be good to include cases for goerli and arbitrum-goerli, since those are valid options for the workflow